### PR TITLE
Inline examples in documentation

### LINKED ISSUE #243: Please more examples ...
Please provide more simple examples regarding to the documentation. The section 'Parsing Expression Types' could be better if the are some simple examples to get into the usage of the expression.

Thank you.

### PR DESCRIPTION
**Pull request by hildjj**
Fixes #243.  This is ready to go in my opinion, but I'm willing to rework it as needed to meet any new discussion or ideas.

### CODE REVIEW

File: docs/documentation.html [hunk 1]
```diff
@@ -649,6 +761,19 @@ <h3 id="grammar-syntax-and-semantics-parsing-expression-types">Parsing Expressio
     results in an array. The matching is greedy, i.e. the parser tries to match
     the expression as many times as possible. Unlike in regular expressions,
     there is no backtracking.</p>
+
+    <div class="example">
+      <div>
+        <div><em>Example:</em> <code>star = "a"*</code></div>
+        <div><em>Matches:</em> <code>"a"</code>, <code>"aaa"</code></div>
+        <div><em>Does not match:</em> </div>
```
Comment (Mingun) [inline docs/documentation.html:769]: May be add an explicitly wording `<nothing (i.e. always match)>` here?
```suggestion
        <div><em>Does not match: &lt;nothing (i.e. always match)&gt;</em> </div>
```

File: docs/documentation.html [hunk 2]
```diff
@@ -666,22 +804,65 @@ <h3 id="grammar-syntax-and-semantics-parsing-expression-types">Parsing Expressio
     <p>Try to match the expression. If the match succeeds, return its match
     result, otherwise return <code>null</code>. Unlike in regular expressions,
     there is no backtracking.</p>
+
+    <div class="example">
+      <div>
+        <div><em>Example:</em> <code>maybe = "a"?</code></div>
+        <div><em>Matches:</em> <code>"a"</code>, <code>""</code></div>
+        <div><em>Does not match:</em> </div>
```
Comment (Mingun) [inline docs/documentation.html:812]: Same here:
```suggestion
        <div><em>Does not match: &lt;nothing (i.e. always match)&gt;</em> </div>
```

**Review by Mingun — APPROVED** [review]
Yeah, I like it! It is impressive work! One minor note: maybe add an explicitly wording that some rules always match?

File: docs/documentation.html (hunk 1 above)
Comment (hildjj) [reply docs/documentation.html]: Nod, I put `(always matches)`, without quotes, and not in `code` format.

**Pull request merged**